File: 02_algorithm/programmers/Level2/H-Index.py
def solution(citations):
    answer = 0
    count_arr = [0] * 10001
    for citation in citations:
        count_arr[citation] += 1
    for idx in range(10000, -1, -1):
        if sum(count_arr[idx : 10001]) >= idx:
            answer = idx
            break
    return answer


print(solution([3, 0, 6, 1, 5]))

File: 06_js/04_algorithm/programmers/Level2/H-Index.js
function solution(citations) {
  let answer = 0
  let countArray = []
  for (let i = 0; i < 10001; ++i) {
      countArray.push(0)
  }
  for (let citation of citations) {
      countArray[citation] += 1
  }
  for (let j = 10000; j > -1; --j) {
      let sumValue = countArray.slice(j, 10001).reduce((a, b) => a + b, 0)
      if (sumValue >= j) {
          answer = j
          break
      }
  }
  return answer
}

console.log(solution([3, 0, 6, 1, 5]))